java/fury-core/src/main/java/org/apache/fury/type/Descriptor.java

```diff
@@ -123,6 +127,7 @@ private Descriptor(Field field, Method readMethod) {
     this.writeMethod = null;
     this.typeRef = null;
     this.furyField = this.field.getAnnotation(FuryField.class);
+    this.nullable = furyField == null || furyField.nullable();
```

Comment:
If furyField is null, nullable should be `!typeRef.isPrimitive()`

Reply:
typeRef is null in this constructor

Reply:
Then it should be `field.getType().isPrimitive()`